File: src/main/org/nlogo/prim/etc/_randomxcor.java
package org.nlogo.prim.etc;

import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

public final strictfp class _randomxcor extends Reporter {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.TYPE_NUMBER);
  }

  @Override
  public Object report(Context context) {
    return report_1(context);
  }

  public double report_1(Context context) {
    double min = world.minPxcor() - 0.5;
    double max = world.maxPxcor() + 0.5;
    return min + context.job.random.nextDouble() * (max - min);
  }
}

File: src/main/prim/etc/_randomxcor.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.core.{ Syntax }
import org.nlogo.nvm.{ Context, Reporter }

class _randomxcor extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(
      ret = Syntax.NumberType)
  override def report(context: Context): java.lang.Double =
    Double.box(report_1(context))
  def report_1(context: Context): Double = {
    val min = world.minPxcor - 0.5
    val max = world.maxPxcor + 0.5
    min + context.job.random.nextDouble * (max - min)
  }
}
